core/tee/tee_svc_storage.c

```diff
@@ -626,6 +627,9 @@ TEE_Result syscall_storage_obj_create(unsigned long storage_id, void *object_id,
 		goto err;
 	}
 
+	/* remove the file if it exists*/
+	if (!filedoesnotexist)
+		tee_file_ops.unlink(file);
```

Comment:
Right, delete and rename is not atomic right now. However, I have no idea how to fix the current implementation of `tee_fs_common_rename` being able to atomically "delete and rename"

@jforissier you are refactoring the FS code and will use new concepts for FS. Do you know if the new way to rename will include "delete if necessary"?